azkaban-exec-server/src/main/java/azkaban/execapp/ProjectCacheCleaner.java

```diff
@@ -152,9 +159,8 @@ public boolean accept(final File dir, final String name) {
       }
     })) {
       if (project.exists() && project.isDirectory()) {
-        ProjectDirectoryMetadata projectDirectoryMetadata = cachedProjects.get(project.toPath());
         if (!projectsUnderDeletion.containsKey(project.toPath())) {
-          projectDirectoryMetadata = fetchMetadataUpdate(project.toPath(), projectDirectoryMetadata);
+          ProjectDirectoryMetadata projectDirectoryMetadata = fetchProjectMetadata(project.toPath());
```

Comment:
Suggestion: Line 170, please add a comment for better readability like,
// end for loop